### PR TITLE
New map import

### PR DESCRIPTION
**Pull request by cm97878**
Overhaul of a bunch of smaller sections of map functionality

- Set map to load from .json generated by auxiliary mapmaker program
- Changed customFunc to a string that triggers a related function, as JSON doesn't play nice with saving functions
- Added a ton of nodes to the map, although most do not have names or descriptions yet
- Changed format of custom nodes to actually accommodate handles

### CODE REVIEW

File: src/components/OvermapPanel.vue [hunk 1]
```diff
@@ -86,11 +88,25 @@ const isConnected = function(node: any): boolean {
...
+            targNode.hidden = false;
+            targNode.data.interactable = true;
+            let secondEdges = getConnectedEdges(targNode.id);
+            secondEdges.forEach(innerEl => {
+                let innerNode = findNode(innerEl.target);
+                if(innerNode) {
+                    innerNode.hidden = false;
+                }
+            })
+        }
```
Comment (Malthusis) [inline src/components/OvermapPanel.vue:103]: Considering this is basically the same code block duplicated, you could make it a function you call where you pass in a given node.

File: src/stores/mapStore.ts [hunk 2]
```diff
@@ -98,241 +100,284 @@ export const useMapStore = defineStore('mapStuff', () => {
...
+    
+
+    
+    //Keeping this to copy data over when we make the new map file
+    // const mapNodes = ref([
+    // {
+    //     id: '1',
+    //     type: 'custom',
+    //     label: 'Home',
+    //     position: { x: 0, y: 0 },
```
Comment (Malthusis) [inline src/stores/mapStore.ts:135]: I think you can safely remove this large comment now.

**Review by Malthusis — APPROVED** [review]
Looks pretty good! 👍

File: src/components/OvermapPanel.vue (hunk 1 above)
Comment (cm97878) [reply src/components/OvermapPanel.vue]: Whole scouting function is still acting weird, realized when I went to double check and make sure the clearing works. It's late, I'll fix whatever this is doing tomorrow

**Comment by cm97878** [comment]
Fixed that nonsense - nodes were set by setNodes( ), which was causing odd reactivity issues, including fitView to not update in certain circumstances.. Changed to binding them in the vue-flow div and everything works as intended.

**Pull request merged**